docs/proposals/27-cloudprofile-bastion-section.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Add Optional Bastion Section To CloudProfile
+gep-number: 27
+creation-date: 2024-05-08
+status: implementable
+authors:
+- "@hebelsan"
+- "@kon-angelo"
+- "@AndreasBurger"
+reviewers:
+- "@rfranzke"
+- "@timuthy"
+---
+
+# GEP-27: Add Optional Bastion Section to CloudProfile
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Prerequisites](#prerequisites)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+
+## Summary
+
+Our plan is to extend the gardener core CloudProfile with a new bastion section that will allow us to create a bastion instance with a specific machine image and machine type that will be referenced in that new section.
+
+## Motivation
+
+We are working on updating the bastion images used in our controllers.
+Some of these machine images were hardcoded in the controller.
+We were using different image "families" in some providers like `Debian`, and in others `GardenLinux`.
+We had "custom" logic per provider to determine the "latest" image and so on.
+
+Our plan is to extend the gardener core CloudProfile with a new bastion section that will allow us to create a bastion instance with a specific machine image and machine type that will be referenced in that new section.
+
+There are some advantages to this approach compared to the current one:
+- Consistently use GardenLinux image where possible
+- Easier to maintain
+- Known systemd and ssh requirements for the images
+
+Drawbacks:
+- Bastion is using larger machine types and larger images
```

Comment:
Can you explain this drawback? With the new section, operators can deliberately choose the machine type for bastion hosts. Why are they larger?

Reply:
Because you can only use machine types that can be used as k8s nodes (per the minimum requirements of a node dictated by gardener) while providers tend to have smaller machines that could fit the bill for a bastion

Reply:
There is a [usable](https://github.com/gardener/gardener/blob/e80591fa667e763a936d544b806689aba96fd951/pkg/apis/core/v1beta1/types_cloudprofile.go#L159) field in the `CloudProfile`. Does it make sense to use this approach to place smaller machines which can be used for bastions?

Reply:
Sounds like an optimisation that we can bring in